From c5c6aab650e7e2ab7d31c0b48da1b0eaca6d4320 Mon Sep 17 00:00:00 2001
From: Zahari Doychev <zahari.doychev@intel.com>
Date: Sun, 1 Dec 2019 14:27:03 +0100
Subject: [PATCH] eth_drv: set ndo_bridge_set/getlink netdev_ops

Set the ndo_bridge_setlink/getlink to switchdev_port_bridge_setlink/getlink
to enable configuration of bridge port settings like mac learning on/off.

Signed-off-by: Zahari Doychev <zahari.doychev@intel.com>
---
 drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
index ecbe56842404..2d20ea0bf4b9 100644
--- a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
+++ b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
@@ -22,6 +22,7 @@
 #include <linux/of_irq.h>
 #include <linux/of_device.h>
 #include <linux/thermal.h>
+#include <net/switchdev.h>
 
 #ifdef CONFIG_USERSPACE_LINK_NOTIFICATION
 #include <net/genetlink.h>
@@ -117,6 +118,8 @@ static const struct net_device_ops ltq_eth_drv_ops = {
 	.ndo_get_stats64	= ltq_get_stats,
 	.ndo_do_ioctl		= ltq_ioctl,
 	.ndo_tx_timeout		= ltq_tx_timeout,
+	.ndo_bridge_setlink	= switchdev_port_bridge_setlink,
+	.ndo_bridge_getlink	= switchdev_port_bridge_getlink,
 };
 
 static struct notifier_block netdevice_notifier = {
